aws/resource_aws_route53_resolver_dnssec_config.go

```diff
@@ -0,0 +1,176 @@
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+	}
+}
+
+func resourceAwsRoute53ResolverDnssecConfigCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).route53resolverconn
+
+	req := &route53resolver.UpdateResolverDnssecConfigInput{
+		ResourceId: aws.String(d.Get("resource_id").(string)),
+		Validation: aws.String(route53resolver.ValidationEnable),
+	}
+
+	log.Printf("[DEBUG] Creating Route53 Resolver DNSSEC config: %#v", req)
+	resp, err := conn.UpdateResolverDnssecConfig(req)
+	if err != nil {
+		return fmt.Errorf("error creating Route53 Resolver DNSSEC config: %s", err)
+	}
+
+	d.SetId(aws.StringValue(resp.ResolverDNSSECConfig.ResourceId))
+
+	err = route53ResolverDnssecConfigWait(conn, d.Id(), d.Timeout(schema.TimeoutCreate),
+		[]string{route53resolver.ResolverDNSSECValidationStatusEnabling},
+		[]string{route53resolver.ResolverDNSSECValidationStatusEnabled})
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsRoute53ResolverDnssecConfigRead(d, meta)
+}
+
+func resourceAwsRoute53ResolverDnssecConfigRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).route53resolverconn
+	ec2Conn := meta.(*AWSClient).ec2conn
+
+	vpc, err := vpcDescribe(ec2Conn, d.Id())
+	if err != nil {
+		return fmt.Errorf("error getting VPC associated with Route53 Resolver DNSSEC config (%s): %s", d.Id(), err)
+	}
+
+	// GetResolverDnssecConfig returns AccessDeniedException if sending a request with non-existing VPC id
+	if vpc == nil {
+		log.Printf("[WARN] VPC associated with Resolver DNSSEC config (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	raw, state, err := route53ResolverDnssecConfigRefresh(conn, d.Id())()
+	if err != nil {
+		return fmt.Errorf("error getting Route53 Resolver DNSSEC config (%s): %s", d.Id(), err)
+	}
+
+	if state == route53ResolverDnssecConfigStatusNotFound || state == route53resolver.ResolverDNSSECValidationStatusDisabled {
+		log.Printf("[WARN] Route53 Resolver DNSSEC config (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	out := raw.(*route53resolver.ResolverDnssecConfig)
+	d.Set("id", out.Id)
+	d.Set("owner_id", out.OwnerId)
+	d.Set("resource_id", out.ResourceId)
+	d.Set("validation_status", out.ValidationStatus)
+
+	return nil
+}
+
+func resourceAwsRoute53ResolverDnssecConfigDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).route53resolverconn
+
+	log.Printf("[DEBUG] Deleting Route53 Resolver DNSSEC config: %s", d.Id())
+	_, err := conn.UpdateResolverDnssecConfig(&route53resolver.UpdateResolverDnssecConfigInput{
+		ResourceId: aws.String(d.Id()),
+		Validation: aws.String(route53resolver.ValidationDisable),
+	})
+	if isAWSErr(err, route53resolver.ErrCodeResourceNotFoundException, "") {
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("error deleting Route53 Resolver DNSSEC config (%s): %s", d.Id(), err)
```

Comment:
[Prefer `%w`](https://github.com/hashicorp/terraform-provider-aws/blob/master/docs/contributing/error-handling.md#wrap-errors).

```suggestion
		return fmt.Errorf("error deleting Route53 Resolver DNSSEC config (%s): %w", d.Id(), err)
```